### PR TITLE
Create new email signup screen for Android

### LINKED ISSUE #7005: Create email signup screen for Android
This needs:
- Textfields for email and password. 
- A button to sign up
- Another button to go to the email login screen
- A back button (to the onboarding)
- A way to show errors

Bind it with the email signup ViewModel

#7003 should be done first.

Designs and full specs: https://www.notion.so/toggl/Designs-wip-d767ebc157c542c790e1bc93eeaf006f#835299ea40344044b2572f356b4f47c1

Android designs are pending, but we can start implementation taking iOS designs as reference and then do the appropriate changes later.

**NOTE:** This issue is part of the `Onboarding refactor` project and it should be merged into the project branch and not into develop.

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This adds the new layout and bindings for the redesigned signup screen. 

### Relationships
Closes #7005 

## Why do we want this?
To look great.

## How is it done?
With the new material theming <3

### Why not another way?
¯\_(ツ)_/¯

### Side effects
None.

## Review hints
You might want to review by commit, but it can be boring.
Please review the layout and the bindings, don't expect the navigation to work properly yet.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs [hunk 1]
```diff
@@ -1,65 +1,48 @@
...
+            welcomeMessage.Text = Shared.Resources.SignUpWelcomeMessage;
+            emailInputLayout.Hint = Shared.Resources.Email;
+            passwordInputLayout.Hint = Shared.Resources.SetPassword;
+            signUpButton.Text = Shared.Resources.SignUp;
+
+            var loginLabelText = new SpannableStringBuilder();
+            loginLabelText.Append(Shared.Resources.HaveAnAccountQuestionMark, new TypefaceSpan(Typeface.SansSerif), SpanTypes.ExclusiveExclusive);
+            loginLabelText.Append(" ");
+            loginLabelText.Append(Shared.Resources.LoginTitle, new TypefaceSpan("sans-serif-medium"), SpanTypes.ExclusiveExclusive);
+            loginLabelText.SetSpan(new UnderlineSpan(), Shared.Resources.HaveAnAccountQuestionMark.Length, loginLabelText.Length(), SpanTypes.ExclusiveExclusive);
```
Comment (pdrj) [inline Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs]:
```suggestion
            loginLabelText.SetSpan(new UnderlineSpan(), Shared.Resources.HaveAnAccountQuestionMark.Length + 1, loginLabelText.Length(), SpanTypes.ExclusiveExclusive);
```

File: Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs [hunk 2]
```diff
@@ -1,65 +1,48 @@
...
     {
-        private Button signupButton;
-
-        private ViewGroup loginCard;
-        private ViewGroup loginSignupContainer;
-        private View googleSignupButton;
-
-        private ImageView countryErrorView;
-
+        private TextInputLayout emailInputLayout;
```
Comment (pdrj) [inline Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs]: In dark mode, the blinking cursor in these fields is barely visible. It should probably be the same color as text (white-ish), not dark. The designs don't spec that, but in the TOS screen in dark mode, the cursor is indeed white.

File: Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs [hunk 3]
```diff
@@ -1,65 +1,48 @@
...
+        private TextInputLayout passwordInputLayout;
```
Comment (pdrj) [inline Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs]: This fields should also not allow multiline (they do ATM).

File: Toggl.Droid/Activities/SignUpActivity.cs [hunk 4]
```diff
@@ -30,80 +30,65 @@ public SignUpActivity(IntPtr javaReference, JniHandleOwnership transfer)
...
 
-            countrySelection.Rx().Tap()
-                .Subscribe(ViewModel.PickCountry.Inputs)
+            loginLabel.Rx()
+                .BindAction(ViewModel.Login)
                 .DisposedBy(DisposeBag);
 
             string signupButtonTitle(bool isLoading)
-                => isLoading ? "" : Shared.Resources.SignUpTitle;
+                => isLoading
```
Comment (pdrj) [inline Toggl.Droid/Activities/SignUpActivity.cs]: In loading state, I can still enter the email field and modify its content. We should not allow that, but if it is tricky to achieve that, it's fine to leave as it is.

Comment (pdrj) [reply Toggl.Droid/Activities/SignUpActivity.cs]: I can also press the 'Loading...' button and get an error saying that user with this email already exists. We should at least disable that button.

File: Toggl.Droid/Resources/layout/SignUpActivity.axml [hunk 5]
```diff
@@ -1,241 +1,145 @@
...
-        <RelativeLayout
-            android:animateLayoutChanges="true"
-            android:focusableInTouchMode="true"
-            android:gravity="center_horizontal"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="40dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginRight="40dp"
+        android:background="?attr/selectableItemBackground"
+        android:clickable="true"
```
Comment (pdrj) [inline Toggl.Droid/Resources/layout/SignUpActivity.axml]: Why is this clickable?

File: Toggl.Droid/Toggl.Droid.csproj [hunk 6]
```diff
@@ -303,11 +303,16 @@
...
+      <HintPath>..\packages\Xamarin.AndroidX.ViewPager2.1.0.0\lib\monoandroid90\Xamarin.AndroidX.ViewPager2.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
     <Reference Include="Xamarin.AndroidX.Work.Runtime, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
       <HintPath>..\packages\Xamarin.AndroidX.Work.Runtime.2.2.0\lib\monoandroid90\Xamarin.AndroidX.Work.Runtime.dll</HintPath>
     </Reference>
     <Reference Include="Xamarin.Google.Android.Material, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
-      <HintPath>..\packages\Xamarin.Google.Android.Material.1.0.0-preview02\lib\monoandroid90\Xamarin.Google.Android.Material.dll</HintPath>
+      <HintPath>..\packages\Xamarin.Google.Android.Material.1.1.0-rc2\lib\monoandroid90\Xamarin.Google.Android.Material.dll</HintPath>
+      <Private>True</Private>
```
Comment (pdrj) [inline Toggl.Droid/Toggl.Droid.csproj]: You know what to do with this 😆

File: Toggl.Droid/Resources/layout/SignUpActivity.axml [hunk 7]
```diff
@@ -1,241 +1,145 @@
...
+        android:fontFamily="sans-serif"
+        android:letterSpacing="0.04"
+        android:textColor="@color/errorText"
+        android:textSize="12sp"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/passwordTextInputLayout"
+        tools:text="Some signup error" />
+
+    <Button
```
Comment (pdrj) [inline Toggl.Droid/Resources/layout/SignUpActivity.axml]: This has the good old problem with overlapping nav.

File: Toggl.Droid/Toggl.Droid.csproj [hunk 8]
```diff
@@ -303,11 +303,16 @@
     <Reference Include="Xamarin.AndroidX.ViewPager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
       <HintPath>..\packages\Xamarin.AndroidX.ViewPager.1.0.0\lib\monoandroid90\Xamarin.AndroidX.ViewPager.dll</HintPath>
     </Reference>
+    <Reference Include="Xamarin.AndroidX.ViewPager2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
+      <HintPath>..\packages\Xamarin.AndroidX.ViewPager2.1.0.0\lib\monoandroid90\Xamarin.AndroidX.ViewPager2.dll</HintPath>
+      <Private>True</Private>
```
Comment (pdrj) [inline Toggl.Droid/Toggl.Droid.csproj]: You know what to do with this 😆

File: Toggl.Droid/packages.config [hunk 9]
```diff
@@ -167,6 +167,7 @@
   <package id="Xamarin.AndroidX.VectorDrawable.Animated" version="1.1.0" targetFramework="monoandroid10.0" />
   <package id="Xamarin.AndroidX.VersionedParcelable" version="1.1.0" targetFramework="monoandroid10.0" />
   <package id="Xamarin.AndroidX.ViewPager" version="1.0.0" targetFramework="monoandroid10.0" />
+  <package id="Xamarin.AndroidX.ViewPager2" version="1.0.0" targetFramework="monoandroid10.0" />
```
Comment (pdrj) [inline Toggl.Droid/packages.config]: Not a big deal, just wondering - why has this been added? Is it a dependency of `Material` package?

File: Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs (hunk 1 above)
Comment (pdrj) [reply Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs]: Screenshot 2020-03-04 at 10 56 57

(that's the reason for the proposed change)

File: Toggl.Droid/Activities/SignUpActivity.cs (hunk 4 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/SignUpActivity.cs]: The overlay in the screen came in hand. It was pretty easy to disable everything.

File: Toggl.Droid/Resources/layout/SignUpActivity.axml (hunk 5 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/layout/SignUpActivity.axml]: ©️ 🍝

File: Toggl.Droid/Resources/layout/SignUpActivity.axml (hunk 7 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/layout/SignUpActivity.axml]: 🙈

File: Toggl.Droid/Toggl.Droid.csproj (hunk 6 above)
Comment (daividssilverio) [reply Toggl.Droid/Toggl.Droid.csproj]: 🙈

File: Toggl.Droid/packages.config (hunk 9 above)
Comment (daividssilverio) [reply Toggl.Droid/packages.config]: yup

File: Toggl.Droid/Toggl.Droid.csproj (hunk 8 above)
Comment (daividssilverio) [reply Toggl.Droid/Toggl.Droid.csproj]: 🙈

File: Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/SignUpActivity.ViewBinds.cs]: this was tricky, but fixed

**Pull request merged**